blockstore/splitstore/splitstore_warmup.go

```diff
@@ -136,14 +58,111 @@ func (s *SplitStore) doWarmup(curTs *types.TipSet) error {
 	if err != nil {
 		return xerrors.Errorf("error saving warm up epoch: %w", err)
 	}
-
 	s.warmupEpoch.Store(int64(epoch))
 
 	// also save the compactionIndex, as this is used as an indicator of warmup for upgraded nodes
 	err = s.ds.Put(s.ctx, compactionIndexKey, int64ToBytes(s.compactionIndex))
 	if err != nil {
 		return xerrors.Errorf("error saving compaction index: %w", err)
 	}
-
 	return nil
 }
+
+// the legacy warmup procedure before we wrote snapshots directly to the hotstore
```

Comment:
TODO put this behind a splitstore config to support the now rare case where people want to start a hot store off of a cold store